File: src/main/java/compiler/block/primitives/IntegerBlock.java
package compiler.block.primitives;

import compiler.block.Block;

public class IntegerBlock extends Block {

    private String type = "int";
    private String value;
    private String name;

    public IntegerBlock(Block superBlock, String name, String value) {
        super(superBlock, false, true);
        this.name = name;
        this.value = value;
    }

    @Override
    public void init() {

    }

    public String getName() {
        return name;
    }

    public String getValue() {
        return value;
    }

    public String getType() {
        return type;
    }

    @Override
    public String getOpeningCode() {
        return "";
    }

    @Override
    public String getBodyCode() {
        return "mv.visitLdcInsn(" + value + ");\n" +
                "mv.visitVarInsn(ISTORE," + getId() + ");\n";
    }

    @Override
    public String getClosingCode() {
        return "";
    }

    public void setType(String type) {
        this.type = type;
    }

    public void setValue(String value) {
        this.value = value;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public String toString() {
        return "int: " + name + " = " + value;
    }
}

File: src/main/scala/compiler/block/primitives/IntegerBlock.scala
package compiler.block.primitives

import compiler.block.Block

class IntegerBlock(superBlockInit: Block, declaration : Boolean, name: String, value: String) extends Block(superBlockInit, false, true) {

  override def init() {}

  override def getName: String = name

  override def getValue: String = value

  override  def getType: String = "int"

  override def getOpeningCode: String = {
    asm.visitLdcInsn("new Integer(" + value + ")") +
      asm.visitVarInsn("ISTORE", id)
  }

  override def getClosingCode: String = {
    ""
  }

  override def toString: String = "int: " + name + " = " + value

}